tools/windows/DatadogAgentInstaller/CustomActions/ConfigCustomActions.cs

```diff
@@ -0,0 +1,374 @@
+                }
+                return true;
+            }
+
+            /// <summary>
+            /// Replaces text from the <paramref name="input"/> string with the replacer functions.
+            /// </summary>
+            /// This function starts with the first pattern and tries to match it
+            /// against the <paramref name="input"/>. If it matches, it will call the replacer function
+            /// associated with this pattern to replace the text in the input and save the index where
+            /// the match occurred.
+            /// It then tries to match the next pattern starting at the index of the previous match, and
+            /// replace the text from the <paramref name="input"/> with the text from its associated
+            /// replacer function.
+            /// If the pattern doesn't have an associated replacer function, it simply saves the index
+            /// of the match and move to the next pattern.
+            /// If a pattern fails to match, an exception is thrown.
+            /// <param name="input">The input string from where to replace the text.</param>
+            /// <param name="propertyValue">The value of the <see cref="PropertyName"/> that will be passed
+            /// to the replacer function.</param>
+            /// <returns>The <paramref name="input"/> string with the text replaced in it.</returns>
+            /// <exception cref="Exception">Throws an exception if any pattern fails to match.</exception>
+            public string DoReplace(string input, string propertyValue)
+            {
+                int matchIndex = 0;
+                foreach (var r in _regexReplacersInOrder)
+                {
+                    var match = r.Regex.Match(input, matchIndex);
+                    if (!match.Success)
+                    {
+                        throw new Exception($"Could not find a match for {r}");
+                    }
+                    if (r.Replacer != null)
+                    {
+                        input = r.Regex.Replace(input, r.Replacer(propertyValue), count: 1, matchIndex);
+                    }
+                    matchIndex = match.Index;
+                }
+
+                return input;
+            }
+        }
+
+        static string FormatProxy(string value, ISession session)
+        {
+            var hostUri = new UriBuilder(value);
+            var proxyConfiguration = new StringBuilder();
+
+            if (!string.IsNullOrEmpty(session.Property("PROXY_USER")))
+            {
+                hostUri.UserName = session.Property("PROXY_USER");
+                if (!string.IsNullOrEmpty(session.Property("PROXY_PASSWORD")))
+                {
+                    hostUri.Password = session.Property("PROXY_PASSWORD");
+                }
+            }
+
+            if (!string.IsNullOrEmpty(session.Property("PROXY_PORT")))
+            {
+                if (!int.TryParse(session.Property("PROXY_PORT"), out var port))
```

Comment:
should we have "better"  port validation (like range) and will it recognize 123abc as 123 or fail parsing?